ament_cmake_python/cmake/ament_python_install_package.cmake

```diff
@@ -42,34 +46,75 @@ function(_ament_cmake_python_install_package package_name)
     set(ARG_PACKAGE_DIR "${CMAKE_CURRENT_LIST_DIR}/${ARG_PACKAGE_DIR}")
   endif()
 
+  if(NOT ARG_VERSION)
+    # Use package.xml version
+    if(NOT _AMENT_PACKAGE_NAME)
+      ament_package_xml()
+    endif()
+    set(ARG_VERSION "${${PROJECT_NAME}_VERSION}")
+  endif()
+
   if(NOT EXISTS "${ARG_PACKAGE_DIR}/__init__.py")
     message(FATAL_ERROR "ament_python_install_package() the Python package "
       "folder '${ARG_PACKAGE_DIR}' doesn't contain an '__init__.py' file")
   endif()
 
-  _ament_cmake_python_register_environment_hook()
+  set(build_dir "${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_python/${package_name}")
 
-  if(NOT PYTHON_INSTALL_DIR)
-    message(FATAL_ERROR "ament_python_install_package() variable "
-      "'PYTHON_INSTALL_DIR' must not be empty")
-  endif()
-  install(
-    DIRECTORY "${ARG_PACKAGE_DIR}/"
-    DESTINATION "${PYTHON_INSTALL_DIR}/${package_name}"
-    PATTERN "*.pyc" EXCLUDE
-    PATTERN "__pycache__" EXCLUDE
+  string(CONFIGURE "\
+from setuptools import setup
+
+setup(
+  name='${package_name}',
+  version='${ARG_VERSION}',
+  packages=['${package_name}'],
```

Comment:
here a list of packages may be desired, instead of only one

Reply:
Yeah, I thought about it, but I didn't want to change the API. Perhaps I can default to `find_packages()` using the `package_dir`.

Reply:
Or maybe, it can be unspecified and the setup.cfg must provide it (this can be specified from either setup.py or setup.cfg).
I'm not sure what happens when you have both though (i.e. if both setup.cfg and setup.py provides `packages`).
If `setup.cfg` setting overrides, then any default in the `setup.py` sounds ok.

> Perhaps I can default to find_packages() using the package_dir.

I know you can specify the `where`, `include` and `exclude` options in `setup.cfg`, from [here](https://setuptools.readthedocs.io/en/latest/userguide/declarative_config.html#options):

```
packages - The find: and find_namespace: directive can be further configured in a dedicated subsection options.packages.find. This subsection accepts the same keys as the setuptools.find_packages and the setuptools.find_namespace_packages function: where, include, and exclude.
```

But I'm not sure if that works when combining a `find_package()` in setup.py and the other options in setup.cfg.
If that does work, defaulting to `find_package()` seems super reasonable (particularly, excluding the `test` package is normal in some of our repos).

Reply:
> Or maybe, it can be unspecified and the setup.cfg must provide it (this can be specified from either setup.py or setup.cfg).

A setup.cfg cannot be mandatory, there's downstream code depending on `ament_python_install_package` as-is.

> But I'm not sure if that works when combining a find_package() in setup.py and the other options in setup.cfg.
If that does work, defaulting to find_package() seems super reasonable (particularly, excluding the test package is normal in some of our repos).

I'll give this a shot.

Reply:
> A setup.cfg cannot be mandatory, there's downstream code depending on ament_python_install_package as-is.

Ah yeah, sounds good!

Reply:
Well, it doesn't seem like I can force `find_packages()` to restrict itself to the `PACKAGE_DIR`. Allowing it to crawl beyond that would break the contract (`ament_python_install_package()` only installs the given package, and there could be more than one in the same ROS package), so I'll stick to one macro one package.

> If setup.cfg setting overrides, then any default in the setup.py sounds ok.

For posterity, it appears this is true. However, it is of limited use here. Since `setup.py` runs in the build tree, it won't find any package unless a `where` argument pointing back to the source tree is provided. I don't have a good solution unless we establish a 1-to-1 correspondence between ROS package and Python package (as there can only be one `setup.cfg` per source tree).